docs/configuration/http_filters/lua_filter.rst

```diff
@@ -0,0 +1,334 @@
+.. _config_http_filters_lua:
+
+Lua
+===
+
+.. attention::
+
+  The Lua scripting HTTP filter is **experimental**. Use in production at your own risk. It is
+  being released for initial feedback on the exposed API and for further development, testing,
+  and verification. This warning will be removed when we feel that the filter has received enough
+  testing and API stability to call it generally production ready.
+
+Overview
+--------
+
+The HTTP Lua filter allows `Lua <https://www.lua.org/>`_ scripts to be run during both the request
+and response flows. `LuaJIT <http://luajit.org/>`_ is used as the runtime. Because of this, the
+supported Lua version is mostly 5.1 with some 5.2 features. See the `LuaJIT documentation
+<http://luajit.org/extensions.html>`_ for more details.
+
+The filter only supports loading Lua code in-line in the configuration. If local filesystem code
+is desired, a trivial in-line script can be used to load the rest of the code from the local
+environment.
```

Comment:
This is probably another place need to think about sandboxing if we want to support multi-tenancy fully.